sql/src/main/java/io/crate/planner/RoutingBuilder.java

```diff
@@ -94,8 +88,8 @@ ReaderAllocations buildReaderAllocations() {
                 }
             }
         }
-        readerAllocations = new ReaderAllocations(indexBaseBuilder.build(), shardNodes, indicesByTable);
-        return readerAllocations;
+        routingListByTable.clear();
```

Comment:
why must this be cleared now? by doing this we'd implicitly make only 1 call to this method working as expected afaik. not saying that this is wrong but it looks a bit odd as it introduce this limitation so I'm curious why that is needed. otherwise we maybe should consider to refactor this logic to get allocated routings passed in or so...